app/reducers/sessions.js

```diff
@@ -0,0 +1,165 @@
+import {
+    NEW_SESSION,
+    SWITCH_SESSION,
+    DELETE_SESSION,
+    UPDATE_CONNECTION,
+    UPDATE_CONFIGURATION,
+    UPDATE_IPC_STATE
+} from '../actions/sessions.js';
+import Immutable from 'immutable';
+import {EMPTY_SESSION} from '../constants/constants';
+
+// see end of file for `list` description
+const INITIAL_STATE = Immutable.Map({
+    list: Immutable.Map({}),
+    sessionSelected: 0,
```

Comment:
Can we move to random strings as IDs instead of numbers? I think it'll be a lot easier to understand and less prone to silly number vs string errors.